src/libraries/System.Reflection.DispatchProxy/src/System/Reflection/DispatchProxyGenerator.cs

```diff
@@ -61,7 +61,14 @@ internal static class DispatchProxyGenerator
         private static readonly ProxyAssembly s_proxyAssembly = new ProxyAssembly();
         private static readonly MethodInfo s_dispatchProxyInvokeMethod = typeof(DispatchProxy).GetMethod("Invoke", BindingFlags.NonPublic | BindingFlags.Instance)!;
         private static readonly MethodInfo s_getTypeFromHandleMethod = typeof(Type).GetRuntimeMethod("GetTypeFromHandle", new Type[] { typeof(RuntimeTypeHandle) })!;
-        private static readonly MethodInfo s_makeGenericMethodMethod = typeof(MethodInfo).GetMethod("MakeGenericMethod", new Type[] { typeof(Type[]) })!;
+        private static readonly MethodInfo s_makeGenericMethodMethod = GetGenericMethodMethodInfo();
+
+        [UnconditionalSuppressMessage("ReflectionAnalysis", "IL2026:RequiresUnreferencedCode",
+            Justification = "Use of the MakeGenericMethod is done in a trim-safe way in DispatchProxyGenerator.")]
```

Comment:
I think it would be best to say _why_ this is trim-safe, instead of just saying "this is trim-safe".

Reply:
Thanks, yes it will be better to do so. How about "Use of the MakeGenericMethod is done in a trim-safe way since the type arguments are known at the proxy method generation time"?

Reply:
Here's the suppression I had at first, before this code was refactored. I think it describes the situation pretty well:

https://github.com/dotnet/runtime/blob/9f1c9132bd6b4213c9f344eecc7ecfd9693c32ca/src/libraries/System.Reflection.DispatchProxy/src/System/Reflection/DispatchProxyGenerator.cs#L149-L151